### ISSUE TITLE
Tax rate calculation on discounted line items

### ISSUE THREAD

**Issue by amazingmaze** [Geta/klarna#174]
Hello,

Using Klarna.Checkout.v3 (4.0.1) and EPiServer.Commerce (14.9.0).

We're getting the tax rate calculations wrong on the data sent to Klarna when we have active line item discounts. Looking into GetPrices extension of LineItems (_Klarna.Common.Extensions.GetPrices_) I can see that its using PlacedPrice when we have prices including tax. Problem here is that the salestax is most likely calculated on the discounted price. 

So, the tax rate is getting calculated using the placed price (without discounts) but with tax thats calculated with a line item discount. 

Summary:
```
var unitPrice = lineItem.PlacedPrice;
var salesTax = _taxCalculator.Service.GetSalesTax(lineItem, market, shipment.ShippingAddress, discountedPrice);
taxRate = AmountHelper.GetAmount(salesTax / (unitPrice - salesTax) * 100); 
```
As an example:
The salestax of our item: 4419.15 SEK  (here it should use tax from PlacedPrice and not the discountedPrice)
Placed price: 25995 SEK
The actual price, discounted: 22095.75 SEK (22095.75 x 0.2 = 4419.15 tax)

4419.15 / (25995 - 4419.15) * 100 = 20.48 (should be 25).

If I switch out placed price to the discounted price its correct:
4419.15 / (22095.75 - 4419.15) * 100 = 25

I'm not the original author of the customer code but I've searched through the code for any custom ITaxCalculators or DefaultTaxCalculators and can't find any. If there's  any custom code thats overriding the sales tax calculator.

Shouldn't the tax rate calculations be based on the line item price minus discount and not the placed price, or do you have any idea what I'm missing here? 

Any help is much appreciated!

/Mattias

**Comment by frederikvig** [comment]
Thanks for reporting this Mattias.

**Comment by frederikvig** [comment]
Hi Mattias,

I have an update package with a fix for the tax rate and discounted amount. Would you be willing to test it out in your solution to see if it solves your problem? My email address is: frederik@getadigital.com.

**Comment by frederikvig** [comment]
Fixed in version 4.1.0.
